src/order/basic.lean

```diff
@@ -192,6 +192,20 @@ end order_dual
 
 namespace strict_mono_incr_on
 
+section dual
+
+variables [preorder α] [preorder β] {f : α → β} {s : set α}
+
+protected lemma dual (H : strict_mono_incr_on f s) :
+  @strict_mono_incr_on (order_dual α) (order_dual β) _ _ f s :=
+λ x hx y hy, H hy hx
+
+protected lemma dual_right (H : strict_mono_incr_on f s) :
```

Comment:
Would `dual_range` or `dual_codomain` be a better name, since I understand this is taking the dual of the output of `f`?

Also, is it worth adding `dual_domain` / `dual_left` as well?

Reply:
Both `dual_right` and `dual_codomain` look good to me. I'd prefer to add trivial lemmas like these on as-needed basis, and I don't need `dual_left`/`dual_domain` now.